### PR TITLE
enable TKGS featuregate validation check for TKGS classy cluster creation

### LINKED ISSUE #2203: Workload cluster creation with cluster class input file fails in TKGS cluster
**Bug description**
`tanzu create cluster -f ccluster.yaml` command with TKGS cluster class based Cluster (Cluster object) input file failing for TKGS use case. 
**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
`tanzu create cluster` with cluster class input file should work fine for TKGS cluster like for AWS cluster.
**Steps to reproduce the bug**
run `tanzu create cluster -f ccluster.yaml` cluster class input file in TKGS management cluster
**Version** (include the SHA if the version is not obvious)
version: v0.21.0-dev
**Environment where the bug was observed (cloud, OS, etc)**
TKGS cluster
**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This PR enables TKGS featuregate validation for TKGS cluster class based cluster creation, we had implemented the featuregate feature validation in TKG CLI but we had disabled featuregate validation because TKGS team was not implemented fully, now TKGS team implemented and enabled the featuregate hence enabling it in TKG CLI.

In TKG CLI, for TGKS cluster class based cluster creation, CLI uses [FeatureGate API](https://pkg.go.dev/github.com/vmware-tanzu/tanzu-framework/pkg/v1/sdk/features/featuregate) to query TKGS supervisor cluster for the feature "vmware-system-tkg-clusterclass" in namespace "vmware-system-tkg", if its not enabled then it throws an error. 

### Which issue(s) this PR fixes

Fixes #2203 

### Describe testing done for PR
Create TKGS Cluster class cluster
Create TKC cluster
Unit test cases

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by anujc25 — APPROVED** [review]
LGTM

File: pkg/v1/tkg/tkgctl/create_cluster_test.go [hunk 1]
```diff
@@ -554,10 +558,70 @@ var _ = Describe("Unit tests for - (Azure) - cluster_azure.yaml as input file fo
...
+			configDir:              testingDir,
+			tkgClient:              tkgClient,
+			tkgConfigReaderWriter:  tkgConfigReaderWriter,
+			tkgConfigUpdaterClient: tkgconfigupdater.New(testingDir, nil, tkgConfigReaderWriter),
+			tkgBomClient:           bomClient,
+			featureGateHelper:      fg,
+		}
+		tkgClient.IsFeatureActivatedReturns(true)
+	})
+	Context("When input file is valid Cluster Class, plan devcc:", func() {
```
Comment (vuil) [inline pkg/v1/tkg/tkgctl/create_cluster_test.go:583]:
```suggestion
	Context("When input file is valid Cluster Class", func() {
```

File: pkg/v1/tkg/tkgctl/create_cluster_test.go [hunk 2]
```diff
@@ -554,10 +558,70 @@ var _ = Describe("Unit tests for - (Azure) - cluster_azure.yaml as input file fo
 	})
 })
 
+var _ = Describe("Unit tests for - TKGS Cluster - cluster_tkgs.yaml as input file for 'tanzu cluster create -f cluster_tkgs.yaml' use case", func() {
```
Comment (vuil) [inline pkg/v1/tkg/tkgctl/create_cluster_test.go:561]: nit: describe at high level what the tests do, but "Unit tests for -" is kinda redundant.

File: pkg/v1/tkg/tkgctl/create_cluster_test.go [hunk 3]
```diff
@@ -645,71 +692,33 @@ var _ = Describe("Unit tests for feature flag (config.FeatureFlagPackageBasedLCM
...
-					featureGateHelper:      fg,
-				}
-				// feature flag (config.FeatureFlagPackageBasedLCM) activated, its clusterclass config input file, but "clusterclass" feature in FeatureGate is enabled,
-				// but throws error for the FeatureGate api, so we expect error here.
-				err := tkgctlClient.CreateCluster(options)
-				// as FeatureGate api throws error, we expect error.
-				Expect(err.Error()).To(ContainSubstring(errorMsg))
-			})
-		*/
+		It("When Feature constants.CCFeature is disabled in featuregate, return error", func() {
```
Comment (vuil) [inline pkg/v1/tkg/tkgctl/create_cluster_test.go:695]: As mentioned before try rephrasing so the test output reads more naturally, 
something like (It("returns error when xxx is disabled"). Same for a number of these test cases.

**Pull request merged**